### PR TITLE
ARS-358 Start page and reg address check page

### PR DESCRIPTION
**Pull request by C-Thomp**
スクリーンショット 0005-01-04 16 33 33

スクリーンショット 0005-01-04 16 32 29

### CODE REVIEW

File: app/controllers/AccountHomeController.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import controllers.actions._
+import javax.inject.Inject
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.AccountHomeView
+
+class AccountHomeController @Inject()(
+                                       override val messagesApi: MessagesApi,
```
Comment (RawToast) [inline app/controllers/AccountHomeController.scala:27]: Looks like scalafmt hasn't been ran on these changes. We need to automate that (or a check) so it's an easy miss 😓

File: app/controllers/RequiredInformationController.scala [hunk 2]
```diff
@@ -16,23 +16,20 @@
...
-import play.api.i18n.{I18nSupport, MessagesApi}
-import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
-import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
-
 import controllers.actions._
 import forms.RequiredInformationFormProvider
-import models.Mode
-import models.UserAnswers
+import models.{NormalMode, UserAnswers}
 import navigation.Navigator
```
Comment (RawToast) [inline app/controllers/RequiredInformationController.scala:22]: I presume IntelliJ is doing this, might just need to ensure it's using scalafmt and not the built in formatter

Comment (C-Thomp) [reply app/controllers/RequiredInformationController.scala:22]: hm i think so

File: app/controllers/AccountHomeController.scala [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                                       identify: IdentifierAction,
+                                       getData: DataRetrievalAction,
+                                       requireData: DataRequiredAction,
+                                       val controllerComponents: MessagesControllerComponents,
+                                       view: AccountHomeView
+                                     ) extends FrontendBaseController with I18nSupport {
+
+  def onPageLoad: Action[AnyContent] = (identify andThen getData andThen requireData) {
```
Comment (RawToast) [inline app/controllers/AccountHomeController.scala:35]: I think this should be: `(identify andThen getData)`, at the moment the user will not have data on this page; although later we will need to handle that case to show any existing applications.

This was probably fine yesterday as the local user had existing data

Comment (RawToast) [reply app/controllers/AccountHomeController.scala]: addressed in https://github.com/hmrc/advance-valuation-rulings-frontend/pull/10/commits/cb871fb415a2e71eb509a3829d668a5970dddd09

**Review by ymabdulahi — APPROVED** [review]

**Pull request merged**